src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py

```diff
@@ -47,7 +47,10 @@ def encrypt_vm(resource_group_name, vm_name,  # pylint: disable=too-many-locals,
                key_encryption_keyvault=None,
                key_encryption_key=None,
                key_encryption_algorithm='RSA-OAEP',
-               volume_type=None):
+               volume_type=None,
+               encrypt_format_all=False,
+               extension_name=None,
+               extension_publisher=None):
```

Comment:
any reason we want to expose this?

Reply:
The latter. This is to allow the dev team and certain partners to check out test versions of the ADE extension.

Reply:
I suggest not to confuse end users, maybe consider other means such as env variables?

Reply:
I remember discussing this with the team as a whole- and we had ended up agreeing on the parameter input approach. If you have some other commands also using environment variables to target test-versions as examples to follow, I would be willing to take a look.

Reply:
Seriously, I don’t think we should do this.  The implementation and testing related details are not meant to be exposed at command arguments to end users. 
I suggest it is the time you can use https://github.com/Azure/azure-cli-extensions. Once it is stabilized after partners' testing you can move the code back to azure-cli